itest/map_test.go

```diff
@@ -443,6 +443,68 @@ func (ts *mapTestSuite) TestMapFromExistingByFd() {
 	ts.Equal(m1, m2)
 }
 
+func (ts *mapTestSuite) TestGetNextKeyString() {
+	// Create map
+	m := &goebpf.EbpfMap{
+		Type:       goebpf.MapTypeHash,
+		KeySize:    8,
+		ValueSize:  8,
+		MaxEntries: 10,
+	}
+	err := m.Create()
+	ts.NoError(err)
+
+	mapData := map[string]string{"key1": "val1", "key2": "val2", "key3": "val3"}
+
+	// Insert items into hash map
+	for key, value := range mapData {
+		err = m.Insert(key, value)
+		ts.NoError(err)
+	}
+	var currentKey string
+	for {
+		nextKey, err := m.GetNextKeyString(currentKey)
+		if err != nil {
+			break
+		}
+		val, err := m.LookupString(nextKey)
+		ts.NoError(err)
+		ts.Equal(mapData[nextKey], string(val))
+		currentKey = nextKey
+	}
```

Comment:
I think this test may not cover the case when first call of `m.GetNextKeyString(currentKey)` returns error.
In this case it will pass, which is wrong..

You may want to maintain one more map of successful lookups and compare it at the end:
```go
result := map[string]string{}
for {
	// ...
	val, err := m.LookupString(nextKey)
	ts.NoError(err)
	result[nextKey] = val
}

ts.Equal(mapData, result)
```

The same for integer based test